.github/workflows/scripts/slack_gem_notifications/cve_methods.rb

```diff
@@ -0,0 +1,31 @@
+# encoding: utf-8
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+require 'time'
+require 'feedjira'
+require 'httparty'
+
+def check_for_cves
+  xml = HTTParty.get('https://rubysec.com/atom.xml').body
+  feed = Feedjira.parse(xml)
+  feed.entries.each do |entry|
+    break if Time.now.utc - entry.updated > 24 * 60 * 60
```

Comment:
It would be good to have the 24 hours value live in a constant. That way Ruby isn't performing the math over and over again on every single feed item.

Reply:
I recommend calculating `Time.now.utc` just once and storing it in a local variable before starting to loop over the feeds. Otherwise, it will keep representing a later and later time as the feed iteration keeps running.

For this exact scenario in which we're expecting to be done in a single second, it's not a big deal. But if iterating was to take multiple minutes and `Time.now` kept advancing, there would be a risk of missing out on data that was 23 hours and 59 minutes old after the iteration was left running for 2 minutes.

Reply:
Super great feedback. I was wondering about the little gap in time from starting to run this file to the end. I like this suggestion a lot. [4052615](https://github.com/newrelic/newrelic-ruby-agent/pull/1292/commits/4052615af1e88f7ead3746b7526c03cdc43c73e6)